core/core_modules/disable.js

```diff
@@ -1,24 +1,52 @@
-var disabled = false;
+// TODO Should send a different message if caused by the counter (should extract all messages into an array)
+// TODO Should be made configurable by sending /disable /counter <counterLimit> (any state is fine)
+// TODO Should flick disabled state using a timer by setting a timer /disable /timer <seconds> (any state is fine)
+// TODO Should be able to reset to default settings /disable /default (any state is fine, only defaults the configuration)
+// TODO Persistent config (counterLimit and disabled state) fairly acheivable by getting and setting properties in exports.platform.modules.disabledConfig
+// TODO Update local var setup after doing the above
+// TODO Should store disable states for each separate thread_id
 
-exports.match = function(event, commandPrefix) {
-	return disabled || event.body === commandPrefix + 'disable';
+var isDisabled = false;
+var prevTimeStamp;
+var counter = 0;
+var counterLimit = 3;
+
+exports.load = function() {
+	prevTimeStamp = Date.now();
+};
+
+exports.match = function(text, commandPrefix) {
+	text += ""; // TODO #PATCH stringifying for handling all kinds of match calls
+	if(text === commandPrefix + 'disable') {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/theprojectshop/Kassy/file/2532146967/issues/source?bid=3282987&fileBranchId=3282995#l20)